raiden/network/resolver/client.py

```diff
@@ -74,7 +74,7 @@ def reveal_secret_with_resolver(
         if response is not None:
             if response.status_code == HTTPStatus.OK:
                 break
-            if response.status_code == HTTPStatus.NOT_FOUND:
+            else:
```

Comment:
@rakanalh please allow me some time to discuss this internally.

Reply:
What are the possible codes the resolver may return?

Reply:
I've been discussing this with offer. Currently it's designed to return 4 status codes:

200 - success, preimage in response body
400 - request was invalid (missing parameter, malformed, etc..)
404 - requested hash was not found
500 - request was fine, but the secret could not be resolved due to internal errors.

My thinking behind the retry logic is that we'd want to keep retrying the resolver endpoint until we get some sort of response indicating that the resolver received and handled our request. If there's an error, treat it as an error like was done previously and continue the flow without the resolver. The status code then would be useful in indicating why the call failed, for debugging purposes for example. `400` or `404` codes as described above would indicate that raiden is doing something wrong, for instance.

More generally speaking, I think it makes sense to intelligently handle all status codes if possible. It's a less fragile approach, especially if using other services as resolvers in the future and knowing.

The one error status code where it would make sense to me to continue retrying is `503 Service Unavailable`, although we're currently not using that. It might be wise to add it here anyway for robustness/completeness sake.